src/PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.hpp

```diff
@@ -0,0 +1,259 @@
+ * \sin\theta\cos\theta\\
+ * F_{\theta\phi} &= B_0\left[\Delta +
+ * \frac{2Mr}{\Sigma^2}(r^4 - a^4)\right]\sin\theta\cos\theta\\
+ * F_{\phi r} &= - B_0\left[r + \frac{M a^2}{\Sigma^2}
+ * (r^2 - a^2\cos^2\theta)(1 + \cos^2\theta)\right]\sin^2\theta.
+ * \f}
+ *
+ * where \f$\Sigma = r^2 + a^2\cos^2\theta\f$ and \f$\Delta = r^2 - 2Mr +
+ * a^2\f$. The associated Eulerian magnetic field is
+ *
+ * \f{align*}
+ * B^r = \frac{F_{\theta\phi}}{\sqrt\gamma},\quad
+ * B^\theta = \frac{F_{\phi r}}{\sqrt\gamma},\quad
+ * B^\phi = \frac{F_{r\theta}}{\sqrt\gamma}.
+ * \f}
+ *
+ * where \f$\gamma = \text{det}(\gamma_{ij})\f$. Wald's solution reproduces a
+ * uniform magnetic field far from the black hole.
+ */
+class BondiHoyleAccretion {
+ public:
+  /// The mass of the black hole.
+  struct BlackHoleMass {
+    using type = double;
+    static constexpr OptionString help = {"The mass of the black hole."};
+  };
+  /// The black hole spin magnitude in units of the black hole mass.
+  struct BlackHoleSpin {
+    using type = double;
+    static constexpr OptionString help = {
+        "The dimensionless black hole spin magnitude."};
+  };
+  /// The rest mass density of the fluid far from the black hole.
+  struct RestMassDensity {
+    using type = double;
+    static constexpr OptionString help = {"The asymptotic rest mass density."};
+  };
+  /// The magnitude of the spatial velocity far from the black hole.
+  struct FlowSpeed {
+    using type = double;
+    static constexpr OptionString help = {
+        "The magnitude of the asymptotic flow velocity."};
+  };
+  /// The strength of the magnetic field.
+  struct MagFieldStrength {
+    using type = double;
+    static constexpr OptionString help = {
+        "The strength of the magnetic field."};
+  };
+  /// The polytropic constant of the fluid.
+  struct PolytropicConstant {
+    using type = double;
+    static constexpr OptionString help = {
+        "The polytropic constant of the fluid."};
+  };
+  /// The polytropic exponent of the fluid.
+  struct PolytropicExponent {
+    using type = double;
+    static constexpr OptionString help = {
+        "The polytropic exponent of the fluid."};
+  };
+
+  using options =
+      tmpl::list<BlackHoleMass, BlackHoleSpin, RestMassDensity, FlowSpeed,
+                 MagFieldStrength, PolytropicConstant, PolytropicExponent>;
+
+  static constexpr OptionString help = {
+      "Axially symmetric accretion on to a Kerr black hole."};
+
+  BondiHoyleAccretion() = default;
+  BondiHoyleAccretion(const BondiHoyleAccretion& /*rhs*/) = delete;
+  BondiHoyleAccretion& operator=(const BondiHoyleAccretion& /*rhs*/) = delete;
+  BondiHoyleAccretion(BondiHoyleAccretion&& /*rhs*/) noexcept = default;
+  BondiHoyleAccretion& operator=(BondiHoyleAccretion&& /*rhs*/) noexcept =
+      default;
+  ~BondiHoyleAccretion() = default;
+
+  BondiHoyleAccretion(BlackHoleMass::type black_hole_mass,
+                      BlackHoleSpin::type black_hole_spin,
```

Comment:
Clarify dimensionless spin, dimensionful spin, or dimensionless spin times mass in name of black_hole_spin